pkg/reporterreporter/report.go

```diff
@@ -30,49 +39,137 @@ type RequestData struct {
 	Data      reporter_v1.NamespaceReport `json:"data"`
 }
 
-func (r *Report) Run(ctx context.Context) error {
-	namespaces, err := r.KubernetesClient.CoreV1().Namespaces().List(meta_v1.ListOptions{})
+func NewReport(slurperURI string, cluster string, reporterClient client.Interface, kubeClient kubernetes.Interface, logger *zap.Logger) ctrl.Server {
+	logger.Info("ReporterReporter configured",
+		zap.String("cluster_id", cluster),
+		zap.String("slurper_uri", slurperURI),
+		zap.String("slurper_path", slurperPath))
+
+	return &report{
+		clusterID:        cluster,
+		httpClient:       util.HTTPClient(),
+		kubernetesClient: kubeClient,
+		logger:           logger,
+		mutator: restclient.NewRequestMutator(
+			restclient.BaseURL(slurperURI),
+			restclient.JoinPath(slurperPath),
+			restclient.Method(http.MethodPost)),
+		reporterClient: reporterClient,
+	}
+}
+
+func isRetriableError(statusCode int) bool {
+	_, ok := map[int]bool{
```

Comment:
This is a set `map[string]interface{} { http.StatusRequestTimeout: interface{}... ` roughly, rather than storing bools.

Reply:
or the sets package we use

Reply:
Done.